### PR TITLE
Phpdoc extractor fix

### PR DESCRIPTION
**Pull request by tobi812**

### CODE REVIEW

File: Extraction/Extractor/JmsExtractor.php [hunk 1]
```diff
@@ -124,9 +123,9 @@ public function extract($reflectionClass, $schema, ExtractionContextInterface $e
         }
     }
 
-    private function extractTypeSchema($type, ExtractionContext $extractionContext)
+    private function extractTypeSchema($type, $schema, ExtractionContextInterface $extractionContext)
```
Comment (newLoki) [inline Extraction/Extractor/JmsExtractor.php]: Doc block is missing

File: Extraction/Extractor/PhpDocOperationExtractor.php [hunk 2]
```diff
@@ -20,6 +20,27 @@ class PhpDocOperationExtractor implements ExtractorInterface
     private $exceptionResponseCodes = [];
 
     /**
+     * Return if the extractor can extract the requested data or not.
+     *
+     * @param $source
```
Comment (newLoki) [inline Extraction/Extractor/PhpDocOperationExtractor.php]: Types are missing for parameters

File: Extraction/Extractor/PhpDocOperationExtractor.php [hunk 3]
```diff
@@ -41,49 +62,36 @@ public function extract($method, $operation, ExtractionContextInterface $extract
...
-            $operation->deprecated = true;
+        if($docBlock->getTagsByName('deprecated')) {
+           $operation->deprecated = true;
         }
 
-        /* @var $throwTag \phpDocumentor\Reflection\DocBlock\Tags\Throws */
         foreach ($docBlock->getTagsByName('throws') as $throwTag) {
-            /** @var Object_ $type */
+            /* @var $throwTag \phpDocumentor\Reflection\DocBlock\Tags\Throws */
+
```
Comment (newLoki) [inline Extraction/Extractor/PhpDocOperationExtractor.php]: I would remove the empty line to make clear this type hint belongs to the line below.

File: Schema/Schema.php [hunk 4]
```diff
@@ -18,55 +18,48 @@ class Schema implements \ArrayAccess
      * @var string
      *
      * @JMS\Type("string")
-     * @JMS\Exclude(if="object.ref !== null")
```
Comment (newLoki) [inline Schema/Schema.php:21]: Just for my understanding, why did you removed this condition?

**Comment by newLoki** [comment]
Only very small things which should be fixed.
So IMO good to go :+1:

File: Schema/Schema.php (hunk 4 above)
Comment (tobi812) [reply Schema/Schema.php:21]: It was introduced in one of the latest master-commits, but unfortunately broke everything for me. That's why i removed it.

**Pull request merged**